### ISSUE TITLE
rstudio can't upload file

### ISSUE THREAD

**Issue by yangzhenhui** [openanalytics/shinyproxy#170]
backend is kubernetes. 

* chrome logs: 
```
Request URL: http://10.60.234.34:24955/app_direct/rstudio/upload
Request Method: POST
Status Code: 503 Service Unavailable
```

* shinyproxy logs:
```
 2019-08-26 09:53:13.502 ERROR 1 --- [   XNIO-2 I/O-1] io.undertow.proxy                        : UT005028: Proxy request to /proxy_endpoint/5d3a3440-a100-454d-977a-ea3402d0741d/upload failed
io.undertow.server.TruncatedResponseException: null
	at io.undertow.client.http.HttpRequestConduit.truncateWrites(HttpRequestConduit.java:711) ~[undertow-core-1.4.22.Final.jar!/:1.4.22.Final]
	at io.undertow.conduits.AbstractFixedLengthStreamSinkConduit.terminateWrites(AbstractFixedLengthStreamSinkConduit.java:256) ~[undertow-core-1.4.22.Final.jar!/:1.4.22.Final]
	at org.xnio.conduits.ConduitStreamSinkChannel.shutdownWrites(ConduitStreamSinkChannel.java:178) ~[xnio-api-3.3.8.Final.jar!/:3.3.8.Final]
	at io.undertow.channels.DetachableStreamSinkChannel.shutdownWrites(DetachableStreamSinkChannel.java:79) ~[undertow-core-1.4.22.Final.jar!/:1.4.22.Final]
	at io.undertow.server.handlers.proxy.ProxyHandler$HTTPTrailerChannelListener.handleEvent(ProxyHandler.java:754) ~[undertow-core-1.4.22.Final.jar!/:1.4.22.Final]
	at io.undertow.server.handlers.proxy.ProxyHandler$ProxyAction$1.completed(ProxyHandler.java:646) [undertow-core-1.4.22.Final.jar!/:1.4.22.Final]
	at io.undertow.server.handlers.proxy.ProxyHandler$ProxyAction$1.completed(ProxyHandler.java:561) [undertow-core-1.4.22.Final.jar!/:1.4.22.Final]
	at io.undertow.client.http.HttpClientExchange.invokeReadReadyCallback(HttpClientExchange.java:212) [undertow-core-1.4.22.Final.jar!/:1.4.22.Final]
	at io.undertow.client.http.HttpClientConnection.initiateRequest(HttpClientConnection.java:410) [undertow-core-1.4.22.Final.jar!/:1.4.22.Final]
	at io.undertow.client.http.HttpClientConnection.sendRequest(HttpClientConnection.java:343) [undertow-core-1.4.22.Final.jar!/:1.4.22.Final]
	at io.undertow.server.handlers.proxy.ProxyHandler$ProxyAction.run(ProxyHandler.java:561) [undertow-core-1.4.22.Final.jar!/:1.4.22.Final]
	at io.undertow.util.SameThreadExecutor.execute(SameThreadExecutor.java:35) [undertow-core-1.4.22.Final.jar!/:1.4.22.Final]
	at io.undertow.server.HttpServerExchange.dispatch(HttpServerExchange.java:815) [undertow-core-1.4.22.Final.jar!/:1.4.22.Final]
	at io.undertow.server.handlers.proxy.ProxyHandler$ProxyClientHandler.completed(ProxyHandler.java:316) [undertow-core-1.4.22.Final.jar!/:1.4.22.Final]
	at io.undertow.server.handlers.proxy.ProxyHandler$ProxyClientHandler.completed(ProxyHandler.java:290) [undertow-core-1.4.22.Final.jar!/:1.4.22.Final]
	at io.undertow.server.handlers.proxy.ProxyConnectionPool.connectionReady(ProxyConnectionPool.java:338) [undertow-core-1.4.22.Final.jar!/:1.4.22.Final]
	at io.undertow.server.handlers.proxy.ProxyConnectionPool.access$900(ProxyConnectionPool.java:61) [undertow-core-1.4.22.Final.jar!/:1.4.22.Final]
	at io.undertow.server.handlers.proxy.ProxyConnectionPool$2.completed(ProxyConnectionPool.java:286) [undertow-core-1.4.22.Final.jar!/:1.4.22.Final]
	at io.undertow.server.handlers.proxy.ProxyConnectionPool$2.completed(ProxyConnectionPool.java:273) [undertow-core-1.4.22.Final.jar!/:1.4.22.Final]
	at io.undertow.client.http.HttpClientProvider.handleConnected(HttpClientProvider.java:156) [undertow-core-1.4.22.Final.jar!/:1.4.22.Final]
	at io.undertow.client.http.HttpClientProvider.access$000(HttpClientProvider.java:51) [undertow-core-1.4.22.Final.jar!/:1.4.22.Final]
	at io.undertow.client.http.HttpClientProvider$2.handleEvent(HttpClientProvider.java:127) [undertow-core-1.4.22.Final.jar!/:1.4.22.Final]
	at io.undertow.client.http.HttpClientProvider$2.handleEvent(HttpClientProvider.java:124) [undertow-core-1.4.22.Final.jar!/:1.4.22.Final]
	at org.xnio.ChannelListeners.invokeChannelListener(ChannelListeners.java:92) [xnio-api-3.3.8.Final.jar!/:3.3.8.Final]
	at org.xnio.nio.WorkerThread$ConnectHandle.handleReady(WorkerThread.java:326) [xnio-nio-3.3.8.Final.jar!/:3.3.8.Final]
	at org.xnio.nio.WorkerThread.run(WorkerThread.java:561) [xnio-nio-3.3.8.Final.jar!/:3.3.8.Final] 
```

**Comment by vpanfilov** [comment]
I can confirm the same error with the default docker backend.

**Comment by vpanfilov** [comment]
Seems to be connected with https://github.com/openanalytics/shinyproxy/issues/140#issuecomment-478387225

Downgrading to Shiny Proxy 2.1.0 fixed it.

**Comment by kunal15145** [comment]
Any update on this issue ?

**Comment by vpanfilov** [comment]
Issue still present on version 2.3.1.

**Comment by LEDfan** [comment]
Hi @yangzhenhui @vpanfilov @kunal15145 this is fixed in [ShinyProxy 2.4.0](https://shinyproxy.io/downloads/#240) .

We extensively tested file upload in rstudio, including large files.
Note that configuring multi-part should no longer be needed.

**Comment by avraam-inside** [comment]
versions 3.0.0 and 3.0.1 contain this problem again... Colleagues, is it difficult to add autotests?
